incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -177,8 +177,9 @@ public MessageConsumer newStream(
 
         final MqttKafkaBindingConfig binding = supplyBinding.apply(routedId);
 
-        final MqttKafkaRouteConfig resolved = binding != null ? binding.resolve(authorization) : null;
 
+        final MqttKafkaRouteConfig resolved = binding != null ? binding.resolve(authorization) : null;
+        kafkaMessagesTopic = binding.messagesTopic();
```

Comment:
No need to store this in a field, as it would be overwritten by a different binding stream anyway, agree?

Reply:
Yes, instead we should store it in the subscribeProxy. I think storing somehow is necessary, as the sending of the kafka begin happens in response to onMqttBegin and not at stream creation, when we can access the binding config.